xarray/test/test_dataarray.py

```diff
@@ -352,6 +352,43 @@ def test_broadcast_equals(self):
         self.assertFalse(a.broadcast_equals(c))
         self.assertFalse(c.broadcast_equals(a))
 
+    def test_notnull_equals(self):
```

Comment:
Yes I'll move some the redundant tests to do with different types of overlap etc. into `test_ops`.